ccm_web/server/src/api/api.interfaces.ts

```diff
@@ -13,7 +13,25 @@ export interface APIErrorData {
   statusCode: number
   message: string
 }
+export interface CanvasSectionBase {
+  id: number
+  name: string
+}
+
+export interface CreateSectionError {
+  failedInput: string
+  message: string
+}
+
+export interface CreateSectionsAPIErrorData {
+  statusCode: number
+  errors: CreateSectionError[]
+}
 
 export function isAPIErrorData (value: unknown): value is APIErrorData {
   return hasKeys(value, ['statusCode', 'message'])
 }
+
+export function isAPICreateSectionErrorData (value: unknown): value is CreateSectionsAPIErrorData {
```

Comment:
This seems useful for more than just creating sections.  It should have a more general name and work with a more general error data type.

Reply:
@pushyamig is in the process of changing that, we've discussed it separately (may be alluded to in other comments).

Reply:
Yeah, I open a different PR https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/157/files, please review that as well so can merge that first and bring in that changes here